recipes/wasmtime/all/conanfile.py

```diff
@@ -60,16 +65,23 @@ def validate(self):
             self.output.warn(msg)
 
         try:
-            self.conan_data["sources"][self.version][self._sources_key][str(self.settings.arch)]
+            self.conan_data["sources"][self.version][self._sources_os_key][str(self.settings.arch)]
         except KeyError:
-            raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
+            raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os are not available")
 
         if (self.settings.compiler, self.settings.os) == ("gcc", "Windows") and self.options.shared:
             # FIXME: https://github.com/bytecodealliance/wasmtime/issues/3168
             raise ConanInvalidConfiguration("Shared mingw is currently not possible")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version][self._sources_key][str(self.settings.arch)], destination=self.source_folder, strip_root=True)
+    def package_id(self):
+        del self.info.settings.compiler.version
+        if self.settings.compiler == "clang":
```

Comment:
Seems like it is dirty hack ??

Reply:
It's a way to alias the packages of gcc and clang.
I couldn't remove `self.info.settings.compiler`.

I argue it's a clean hack :stuck_out_tongue:

Reply:
I would say to use compatible compilers instead: https://docs.conan.io/en/latest/creating_packages/define_abi_compatibility.html#compatible-compilers

Reply:
I've done it this way to avoid creating duplicate packages.
Packages with compiler=gcc and compiler=clang will be identical, because nothing is built.

If I understand compatible compiler correctly, then a package for gcc and clang will be created.

Reply:
I actually used a custom package id: see https://docs.conan.io/en/latest/creating_packages/define_abi_compatibility.html#defining-a-custom-package-id

So not a hack after all. It's documented.

Reply:
Compatible packages would require me to list all gcc/clang versions that are compatible which does not exist, or will be always incomplete.

Reply:
Compatible compilers works, but it will cause c3i to generate a package for gcc and clang, which are the same.
So I'd prefer to keep the `package_id` method as it is now.

Reply:
Right, it will generate for both clang and gcc. Indeed using same pkg id will save cci resources.